utils/techutils/techutils_test.go

```diff
@@ -484,19 +507,33 @@ func TestGetTechInformationFromWorkingDir(t *testing.T) {
 			name:                 "nugetTest",
 			tech:                 Nuget,
 			requestedDescriptors: map[Technology][]string{},
+			techProvidedByUser:   false,
 			expected:             map[string][]string{"dir": {filepath.Join("dir", "project.sln"), filepath.Join("dir", "sub1", "project.csproj")}},
 		},
 		{
 			name:                 "dotnetTest",
 			tech:                 Dotnet,
 			requestedDescriptors: map[Technology][]string{},
+			techProvidedByUser:   false,
 			expected:             map[string][]string{"dir": {filepath.Join("dir", "project.sln"), filepath.Join("dir", "sub1", "project.csproj")}},
 		},
+		{
+			// When tech is provided by the user we detect technology by indicator and descriptors and not just by indicator
+			name:                 "yarnTestWithProvidedTechFromUser",
+			tech:                 Yarn,
+			requestedDescriptors: make(map[Technology][]string),
+			techProvidedByUser:   true,
+			expected: map[string][]string{
+				"dir4": {filepath.Join("dir4", "package.json")},
+				"dir3": {filepath.Join("dir3", "package.json")},
+				"dir":  {filepath.Join("dir", "package.json")},
+			},
+		},
```

Comment:
add another tests with `techProvidedByUser=true` (i.e. tech requested by the user), not for Yarn to make sure it does not break other tech when requested